addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -618,276 +606,692 @@ private void onDisconnected() {
-            if (thingHandler instanceof ZoneMinderThingMonitorHandler) {
-                try {
-                    thingHandler.updateAvaliabilityStatus(connection);
-                } catch (Exception ex) {
-                    logger.debug("{}: Failed to call 'updateAvailabilityStatus()' for '{}'", getLogIdentifier(),
-                            thingHandler.getThing().getUID());
-                }
-            }
-
+            statusDescription = "General error occurred (Check log)";
+            updateBridgeStatus(newStatus, statusDetail, statusDescription, true);
         }
 
     }
 
-    protected void updateBridgeStatus(ThingStatus newStatus, ThingStatusDetail statusDetail, String statusDescription) {
+    protected void updateBridgeStatus(ThingStatus newStatus, ThingStatusDetail statusDetail, String statusDescription,
+            boolean updateConnection) {
         ThingStatusInfo curStatusInfo = thing.getStatusInfo();
         String curDescription = StringUtils.isBlank(curStatusInfo.getDescription()) ? ""
                 : curStatusInfo.getDescription();
 
         // Status changed
         if ((curStatusInfo.getStatus() != newStatus) || (curStatusInfo.getStatusDetail() != statusDetail)
-                || (curDescription != statusDescription)) {
-
-            // if (thing.getStatus() != newStatus) {
-            logger.info("{}: Bridge status changed from '{}' to '{}'", getLogIdentifier(), thing.getStatus(),
-                    newStatus);
-
-            if ((newStatus == ThingStatus.ONLINE) && (curStatusInfo.getStatus() != ThingStatus.ONLINE)) {
-                try {
-                    setBridgeConnectionStatus(true);
-                    onConnected();
-                } catch (IllegalArgumentException e) {
-                    // Just ignore that here
-                }
-            } else if ((newStatus == ThingStatus.OFFLINE) && (curStatusInfo.getStatus() != ThingStatus.OFFLINE)) {
-                try {
-                    setBridgeConnectionStatus(false);
-                    onDisconnected();
-                } catch (IllegalArgumentException e) {
-                    // Just ignore that here
-                }
-
+                || (!curDescription.equals(statusDescription))) {
+            if (!curStatusInfo.getStatus().equals(newStatus)) {
+                logger.info("{}: context='updateBridgeStatus' Bridge status changed from '{}' to '{}'",
+                        getLogIdentifier(), thing.getStatus(), newStatus);
             }
+
             // Update Status correspondingly
             if ((newStatus == ThingStatus.OFFLINE) && (statusDetail != ThingStatusDetail.NONE)) {
                 updateStatus(newStatus, statusDetail, statusDescription);
+
+                forcedPriority = RefreshPriority.UNKNOWN;
+                if (updateConnection) {
+                    try {
+                        setConnected(false);
```

Comment:
If this method is not a setter, which is suggested by the number of exceptions it throws, it might be better to call it `disconnect`. Because now it confuse me as the reader